testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java

```diff
@@ -462,48 +429,106 @@ public void reloadAdminOnly(String host) throws IOException, InterruptedExceptio
      *
      * @param host the host to use for the request. Cannot be {@code null}
-     * @param adminOnly if {@code true}, the Host Controller will be reloaded in admin-only mode
-     * @param restartServers if {@code true}, servers will be restarted
-     * @param waitForServers if {@code true}, wait for Servers reload to complete
-     * @param timeout maximum time to wait for the Host Controller and Servers reload to complete, in milliseconds
-     *
+     * @param parameters {@link DomainLifecycleUtil.ReloadParameters} to configure reload
      */
-    private void reload(String host, Boolean adminOnly, Boolean restartServers, Boolean waitForServers, Long timeout) throws IOException, InterruptedException, TimeoutException {
+    public void reload(String host, ReloadParameters parameters) throws IOException, InterruptedException, TimeoutException {
         ModelNode op = new ModelNode();
         op.get(ModelDescriptionConstants.ADDRESS).add(ModelDescriptionConstants.HOST, host);
         op.get(ModelDescriptionConstants.OP).set("reload");
-        if (adminOnly != null) {
-            op.get("admin-only").set(adminOnly);
+        if (parameters.adminOnly != null) {
+            op.get("admin-only").set(parameters.adminOnly);
         }
-        if (restartServers != null) {
-            op.get(ModelDescriptionConstants.RESTART_SERVERS).set(restartServers);
+        if (parameters.restartServers != null) {
+            op.get(ModelDescriptionConstants.RESTART_SERVERS).set(parameters.restartServers);
         }
         long startupTimeout;
-        if (timeout == null) {
+        if (parameters.timeout == null) {
             startupTimeout = configuration.getStartupTimeoutInSeconds();
         } else {
-            startupTimeout = timeout;
+            startupTimeout = parameters.timeout;
         }
 
         executeAwaitConnectionClosed(op);
         // Try to reconnect to the hc
         connect();
         awaitHostController(System.currentTimeMillis(), startupTimeout);
         // check that the servers are up
-        if (adminOnly == null || !adminOnly) {
-            if (waitForServers == null || waitForServers) {
+        if (parameters.adminOnly == null || !parameters.adminOnly) {
+            if (parameters.waitForServers == null || parameters.waitForServers) {
                 awaitServers(System.currentTimeMillis(), startupTimeout);
             }
         }
     }
 
+    /**
+     * ReloadParameters object for the DomainLifecycleUtil reload
+     */
+    public static class ReloadParameters {
+
+        Boolean adminOnly = false;
+
+        Boolean restartServers = null;
+
+        Boolean waitForServers = null;
+
+        Long timeout = null;
+
+        /**
+         * Sets the adminOnly
+         *
+         * @param adminOnly if {@code true}, the Host Controller will be reloaded in admin-only mode
+         * @return this ReloadParameters object
```

Comment:
The Java doc would need to be completed to describe the other cases. I mean, if it is true, it gets reloadaed with admin only, but what happends if it is null or false? What is the difference bewtween them?

This applies to all the parameters.

Looking at it now, I think we could even using just primitives and avoid the ambiguity between null/false, but in that case, we have to review all the callers since, for example, for restart servers, the default value at management operation level is `true`. So callers should be changed to invoke the reload with the correct parameters yo keep the semantic.

@lvydra what do you think? Would it be simpler to use primitives? in any case, the Java doc needs fixing, as it is now, it is a bit ambiguous.

Reply:
Hi @yersan, thanks for the review.

I have updated the Java doc.

Using primitives seems to be more straightforward and it makes the reload method cleaner for sure. Keeping default values on ReloadParameters the same as default values on the management operation level should ensure consistency, if I'm not mistaken.